core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -155,8 +168,12 @@ public Maven(String targets,String name, String pom, String properties, String j
         this(targets, name, pom, properties, jvmOptions, usePrivateRepository, null, null);
     }
     
-    @DataBoundConstructor
     public Maven(String targets,String name, String pom, String properties, String jvmOptions, boolean usePrivateRepository, SettingsProvider settings, GlobalSettingsProvider globalSettings) {
+        this(targets, name, pom, properties, jvmOptions, usePrivateRepository, settings, globalSettings, true);
```

Comment:
I prefer to leave it as it is. None of the old constructors that got "extended" was deprecated.

Ad `@DataBoundSetter`: There are 9 fields in the class:

- 5 of them is final (some have defaults set in constructor overloads)
- 3 can be set by constructor and updated by setter (one of them directly!)
- if we use `@DataBoundSetter` there will be one filed that can not be set by constructor but only using setter.

I am trying not to make it any more messy then it already is.